e2etests/pkg/infra/routers.go

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package infra
+
+import (
+	"fmt"
+
+	"github.com/openperouter/openperouter/e2etests/pkg/frr"
+)
+
+const (
+	ClabPrefix = "clab-kind-"
+	KindLeaf   = ClabPrefix + "leafkind"
+	LeafA      = ClabPrefix + "leafA"
+	LeafB      = ClabPrefix + "leafB"
+)
+
+var (
+	KindLeafContainer = frr.Container{
+		Name:       KindLeaf,
+		ConfigPath: "leafkind",
+	}
+	LeafAContainer = frr.Container{
+		Name:       LeafA,
+		ConfigPath: "leafA",
+	}
+
+	LeafBContainer = frr.Container{
+		Name:       LeafB,
+		ConfigPath: "leafB",
+	}
+)
+
+var links linksForRouter
+
+func init() {
+	links = linksForRouter{
+		nodes: map[string]node{},
+	}
+	links.Add("clab-kind-leafkind", "pe-kind-control-plane", "192.168.11.2", "192.168.11.3")
+	links.Add("clab-kind-leafkind", "pe-kind-worker", "192.168.11.2", "192.168.11.4")
+	links.Add("clab-kind-leafkind", "clab-kind-spine", "192.168.1.5", "192.168.1.4")
+	links.Add("clab-kind-leafA", "clab-kind-spine", "192.168.1.1", "192.168.1.0")
+	links.Add("clab-kind-leafB", "clab-kind-spine", "192.168.1.3", "192.168.1.2")
+	links.Add("clab-kind-leafA", "clab-kind-hostA_red", "192.168.20.1", "192.168.20.2")
+	links.Add("clab-kind-leafA", "clab-kind-hostA_blue", "192.168.21.1", "192.168.21.2")
+	links.Add("clab-kind-leafB", "clab-kind-hostB_red", "192.169.20.1", "192.169.20.2")
+	links.Add("clab-kind-leafB", "clab-kind-hostB_blue", "192.169.21.1", "192.169.21.2")
+}
+
+type linksForRouter struct {
+	nodes map[string]node
+}
+
+func NeighborIP(from, to string) (string, error) {
+	fromNeighbors, ok := links.nodes[from]
+	if !ok {
+		return "", fmt.Errorf("node %s not found", from)
+	}
+	if fromNeighbors.neighs == nil {
+		return "", fmt.Errorf("node %s has no neighbors", from)
+	}
+	toIP, ok := fromNeighbors.neighs[to]
+	if !ok {
+		return "", fmt.Errorf("node %s has no neighbor %s", from, to)
+	}
+	return toIP, nil
+}
+
+func (l *linksForRouter) Add(from, to, addressFrom, addressTo string) {
```

Comment:
nit: the from/to naming is a bit confusing, as the mapping is bidirectional, maybe we should just rename it to A/B?

Reply:
renamed first / second